### ISSUE TITLE
Feature: Overriding a Restyler without disabling others

### ISSUE THREAD

**Issue by iphydf** [restyled-io/restyled.io#197]
I'd like to pass 1 flag to astyle, but I don't want to then also have to list every other restyler manually. I want all of them to be enabled unless I explicitly disable one.

**Comment by pbrisbin** [comment]
Thank you for this request, I absolutely agree that's how it should work. The only thing holding me back is I need to change the configuration format somehow to support it, and I've been dragging my feet on figuring out how to best do that without breaking existing configs. Hopefully I'll come up with something soon!

**Comment by iphydf** [comment]
Could it be:
```yaml
---
restylers:
  include:
    - astyle:
        arguments: ["..."]
  exclude:
    - clang-format
```

**Comment by pbrisbin** [comment]
Well, it's a little tricky because I actually like the current syntax, I just wish it had the behavior you describe.

```yaml
# Example: override astyle and still run all default Restylers
restylers:
  - astyle:
      arguments: [...]

# Example: override astyle, still run all default Restylers, but not clang-format
restylers:
  - astyle:
      arguments: [...]
  - clang-format:
      enabled: false
```

So that's really what I _want_ the configuration format to be, but I'm hesitant to just make it so since there are lots of users in the wild for which that could cause a lot of new Restylers to run (and possibly error).

Also, what happens for users who do want tight control over what Restylers run when I add a new default Restyler? It will run until they add it to their configuration with `enabled:false`. That seems bad.

What about this:

```yaml
# Example *only* run astyle (current behavior in production)
restylers:
  - astyle

# Example: override astyle and still run all default Restylers
overrides:
  - astyle:
      arguments: [...]

# Example: override astyle, still run all default Restylers, but not clang-format
overrides:
  - astyle:
      arguments: [...]
  - clang-format:
      enabled: false
```

Users that want explicit control should use the key `restylers`, so-named because it indicates exactly and literally what `restylers` to run. Users that want to change some defaults but keep others use `overrides`, so-named because you're just overriding as _aspect_ of the Restylers.

Now the issue is duplicate functionality in `restylers` and `overrides` (either could be used to adjust how a given Restyler runs). This is solvable in the code (sharing a parser, etc), but it presents some conceptual confusion, IMO.

**Comment by iphydf** [comment]
I think it's reasonable to have a migration period where behaviour of an API changes. In Bazel for example, there is `tools`, which you pass to `genrule` (generic build rule) that has some semantics. The semantics they really want for it are different and incompatible, so they introduced `exec_tools` so people can migrate. At some point, `tools` will be deprecated, and everybody will be asked to migrate. A few versions later, `tools` semantics will be set to `exec_tools`, and `exec_tools` will be deprecated. A few versions later again, `exec_tools` will be deleted. It gives a lot of time to users to migrate to the new semantics.

In this case, if users would like to disable all restylers that aren't explicitly configured, they could say:

```yaml
overrides:
  - astyle
  - "*":
      enabled: false

# or
overrides:
  - astyle:
      enabled: true
  - "*":
      enabled: false
```

Either of these would work. You could, if you wanted to, pose this as a migration path to new semantics where `overrides` goes through the same (lengthy) migration period as `exec_tools`. Alternatively, I myself would be a happy user of just `overrides` and ignore the fact that `restylers` exists.

**Comment by pbrisbin** [comment]
That wildcard key,

```yaml
- "*":
```

Is a *phenomenal* idea! Thank you for that. I think that solves the concerns that were holding me back. My time is a bit crunched right now, so I can't commit to any concrete _when_, but I consider this pretty high priority so when I can commit time to Restyled, it will likely be for this.

**Comment by pbrisbin** [comment]
OK, should be actually closed now with restyled-io/restyler#100. Documentaiton here: https://github.com/restyled-io/restyled.io/wiki/Configuring-Restyled#wildcard
